src/core/gpu_vk.c

```diff
@@ -3381,13 +3395,31 @@ static gpu_memory* allocate(gpu_memory_type type, VkMemoryRequirements info, VkD
   };
 
   uint32_t blockSize = blockSizes[type];
-  uint32_t cursor = ALIGN(allocator->cursor, info.alignment);
+  ASSERT(blockSize <= (GPU_ALLOC_PAGES_LENGTH * GPU_PAGE_SIZE), "Block size larger than allocator can handle") return NULL;
+
+  uint32_t requiredPages = ALIGN(info.size, GPU_PAGE_SIZE) / GPU_PAGE_SIZE;
+
+  if (allocator->block) {
```

Comment:
This could say `&& info.size <= blockSize` so that "oversized blocks" (resources bigger than block size) can avoid needlessly searching through every block, when we know they're  not going to fit.

Reply:
This should be obviated by the fact I now only let allocators point to blocks that can have multiple allocations. So, it won't scan over an oversized block, because `allocator->block` will be `NULL`.